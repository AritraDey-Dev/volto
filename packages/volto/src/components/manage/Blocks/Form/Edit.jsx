import React, { Component } from 'react';

import { SidebarPortal } from '@plone/volto/components';
import { Form, BlockDataForm } from '@plone/volto/components/manage/Form';

import FormSchema from './schema';

class Edit extends Component {
  render() {
    const schema = FormSchema(this.props);

    return (
      <>
        <Form
          schema={{
            fieldsets: [
              {
                behavior: 'plone',
                fields: ['schema'],
                id: 'default',
                title: 'Default',
              },
            ],
            properties: {
              schema: {
                description: '',
                factory: 'Text',
                title: 'Schema',
                type: 'string',
                widget: 'schema',
              },
            },
            required: [],
            title: 'Form',
            type: 'object',
          }}
          formData={{
            schema: {
              fieldsets: [
                {
                  behavior: 'plone',
                  fields: ['name', 'from', 'subject', 'message'],
                  id: 'default',
                  title: 'Default',
                },
              ],
              properties: {
                name: {
                  description: 'Please enter your full name',
                  factory: 'Text',
                  title: 'Name',
                  type: 'string',
                  widget: 'text',
                },
                from: {
                  description: 'Please enter your e-mail address',
                  factory: 'Text',
                  title: 'From',
                  type: 'string',
                  widget: 'text',
                },
                subject: {
                  description: '',
                  factory: 'Text',
                  title: 'Subject',
                  type: 'string',
                  widget: 'text',
                },
                message: {
                  description: 'Please enter the message you want to send',
                  factory: 'Text',
                  title: 'Message',
                  type: 'string',
                  widget: 'text',
                },
              },
              required: [],
              title: 'Form',
              type: 'object',
            },
          }}
          hideActions
        />

        <SidebarPortal selected={this.props.selected}>
          <BlockDataForm
            schema={schema}
            title={schema.title}
            onChangeField={(id, value) => {
              this.props.onChangeBlock(this.props.block, {
                ...this.props.data,
                [id]: value,
              });
            }}
            onChangeBlock={this.props.onChangeBlock}
            formData={this.props.data}
            block={this.props.block}
            navRoot={this.props.navRoot}
            contentType={this.props.contentType}
          />
        </SidebarPortal>
      </>
    );
  }
}

export default Edit;
