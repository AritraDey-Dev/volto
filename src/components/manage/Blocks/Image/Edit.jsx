/**
 * Edit image block.
 * @module components/manage/Blocks/Image/Edit
 */

import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import { ImageSidebar, SidebarPortal } from '@plone/volto/components';
import { withBlockExtensions } from '@plone/volto/helpers';
import { flattenToAppURL, isInternalURL } from '@plone/volto/helpers';

import MediaWidget from '@plone/volto/components/manage/Widgets/MediaSelectWidget';

/**
 * Edit image block function.
 * @function Edit
 */
const Edit = (props) => {
  const { data, block, onChangeBlock, editable, selected } = props;
  return (
    <div
      className={cx(
        'block image align',
        {
          center: !Boolean(data.align),
        },
        data.align,
      )}
    >
      {data.url ? (
        <img
          className={cx({
            'full-width': data.align === 'full',
            large: data.size === 'l',
            medium: data.size === 'm',
            small: data.size === 's',
          })}
          src={
            isInternalURL(data.url)
              ? // Backwards compat in the case that the block is storing the full server URL
                (() => {
                  if (data.size === 'l')
                    return `${flattenToAppURL(data.url)}/@@images/image`;
                  if (data.size === 'm')
                    return `${flattenToAppURL(
                      data.url,
                    )}/@@images/image/preview`;
                  if (data.size === 's')
                    return `${flattenToAppURL(data.url)}/@@images/image/mini`;
                  return `${flattenToAppURL(data.url)}/@@images/image`;
                })()
              : data.url
          }
          alt={data.alt || ''}
        />
      ) : (
        <div>
          {editable && (
            <MediaWidget
              inline
              id="url"
              title="Source"
              onChange={(id, value) => {
                onChangeBlock(block, {
                  ...data,
                  [id]: value,
                });
              }}
            />
          )}
        </div>
      )}
      <SidebarPortal selected={selected}>
        <ImageSidebar {...props} />
      </SidebarPortal>
    </div>
  );
};

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
Edit.propTypes = {
  selected: PropTypes.bool.isRequired,
  block: PropTypes.string.isRequired,
  data: PropTypes.objectOf(PropTypes.any).isRequired,
  content: PropTypes.objectOf(PropTypes.any).isRequired,
  onChangeBlock: PropTypes.func.isRequired,
};

export default withBlockExtensions(Edit);
