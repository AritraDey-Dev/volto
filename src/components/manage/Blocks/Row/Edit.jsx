import React, { useState } from 'react';
import { pickBy } from 'lodash';
import { BlocksForm, SidebarPortal, Icon } from '@plone/volto/components';
import PropTypes from 'prop-types';
import { Button } from 'semantic-ui-react';
import RowData from './Data';
import EditBlockWrapper from './EditBlockWrapper';
import { useIntl } from 'react-intl';
import { v4 as uuid } from 'uuid';
import cx from 'classnames';

import TemplateChooser from '@plone/volto/components/manage/TemplateChooser/TemplateChooser';

import addSVG from '@plone/volto/icons/add.svg';
import configSVG from '@plone/volto/icons/configuration.svg';

import templates from './templates';

import config from '@plone/volto/registry';

function getBlockConfig(type) {
  return config.blocks.blocksConfig?.[type];
}

function initBlocksForm() {
  return {
    blocks: {},
    blocks_layout: {
      items: [],
    },
  };
}

const RowEdit = (props) => {
  const {
    block,
    data,
    onChangeBlock,
    onChangeField,
    pathname,
    selected,
    manage,
  } = props;

  const intl = useIntl();
  const metadata = props.metadata || props.properties;
  const isInitialized = data?.blocks && data?.blocks_layout;
  const properties = isInitialized ? data : initBlocksForm();
  const blockConfig = getBlockConfig(data['@type']);
  const blocksConfig =
    config.blocks.blocksConfig[data['@type']].blocksConfig ||
    props.blocksConfig;
  const allowedBlocks = blockConfig.allowedBlocks;
  const maxRowLength = blockConfig.maxRowLength || 8;

  const [selectedBlock, setSelectedBlock] = useState(
    properties.blocks_layout.items[0],
  );

  // Seems never applies...
  // React.useEffect(() => {
  //   if (
  //     isEmpty(data_blocks) &&
  //     properties.blocks_layout.items[0] !== selectedBlock
  //   ) {
  //     setSelectedBlock(properties.blocks_layout.items[0]);
  //     onChangeBlock(block, {
  //       ...data,
  //       data: properties,
  //     });
  //   }
  // }, [onChangeBlock, properties, selectedBlock, block, data, data_blocks]);

  const blockState = {};

  const onAddNewBlock = () => {
    const newuuid = uuid();
    const type = allowedBlocks?.length === 1 ? allowedBlocks[0] : null;
    const newFormData = {
      ...data,
      blocks: {
        ...data.blocks,
        [newuuid]: {
          ...(type && { '@type': type }),
        },
      },
      blocks_layout: {
        items: [...data.blocks_layout.items, newuuid],
      },
    };
    if (data.blocks_layout.items.length < maxRowLength) {
      onChangeBlock(block, {
        ...newFormData,
      });
    }
  };

  const onSelectTemplate = (templateIndex) => {
    const resultantTemplates =
      allowedBlocks?.length === 1 ? templates(allowedBlocks[0]) : templates();
    onChangeBlock(block, {
      ...data,
      ...resultantTemplates(intl)[templateIndex].blocksData,
    });
  };

  const allowedBlocksConfig = pickBy(blocksConfig, (value, key) =>
    allowedBlocks.includes(key),
  );

  const direction = data['@type'] === 'row' ? 'horizontal' : 'vertical';

  const columnsLength = (isInitialized && data.blocks_layout.items.length) || 0;

  return (
    <div
      className={cx({
        one: columnsLength === 1,
        two: columnsLength === 2,
        three: columnsLength === 3,
        four: columnsLength === 4,
        rows: true,
      })}
    >
      {data.headline && <h2 className="headline">{data.headline}</h2>}

      {selected && (
        <div className="toolbar">
          <Button.Group>
            <Button
              aria-label={`Add row element`}
              icon
              basic
              disabled={data?.blocks_layout?.items?.length >= maxRowLength}
              onClick={(e) => onAddNewBlock()}
            >
              <Icon name={addSVG} size="24px" />
            </Button>
          </Button.Group>
          <Button.Group>
            <Button
              aria-label={`Row block settings`}
              icon
              basic
              onClick={(e) => {
                e.stopPropagation();
                setSelectedBlock();
                props.setSidebarTab(1);
              }}
            >
              <Icon name={configSVG} size="24px" />
            </Button>
          </Button.Group>
        </div>
      )}
      {!isInitialized && (
        <TemplateChooser
          templates={
            allowedBlocks?.length === 1
              ? templates(allowedBlocks[0])
              : templates()
          }
          onSelectTemplate={onSelectTemplate}
        />
      )}

      <BlocksForm
        metadata={metadata}
        properties={properties}
        direction={direction}
        manage={manage}
        isMainForm={false}
        selectedBlock={selected ? selectedBlock : null}
        blocksConfig={allowedBlocksConfig}
        title={data.placeholder}
        onSelectBlock={(id) => {
          setSelectedBlock(id);
        }}
        onChangeFormData={(newFormData) => {
          onChangeBlock(block, {
            ...data,
            ...newFormData,
          });
        }}
        onChangeField={(id, value) => {
          if (['blocks', 'blocks_layout'].indexOf(id) > -1) {
            blockState[id] = value;
            onChangeBlock(block, {
              ...data,
              ...blockState,
            });
          } else {
            onChangeField(id, value);
          }
        }}
        pathname={pathname}
      >
        {({ draginfo }, editBlock, blockProps) => (
          <EditBlockWrapper draginfo={draginfo} blockProps={blockProps}>
            {editBlock}
          </EditBlockWrapper>
        )}
      </BlocksForm>
      <SidebarPortal selected={selected && !selectedBlock}>
        <RowData {...props}></RowData>
      </SidebarPortal>
    </div>
  );
};

RowEdit.propTypes = {
  block: PropTypes.string.isRequired,
  onChangeBlock: PropTypes.func.isRequired,
  pathname: PropTypes.string.isRequired,
  selected: PropTypes.bool.isRequired,
  manage: PropTypes.bool.isRequired,
};

export default RowEdit;
