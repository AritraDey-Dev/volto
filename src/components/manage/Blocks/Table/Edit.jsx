/**
 * Edit text block.
 * @module components/manage/Blocks/Title/Edit
 */

import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { compose } from 'redux';
import { map, remove } from 'lodash';
import { Button, Segment, Table, Form } from 'semantic-ui-react';
import { Portal } from 'react-portal';
import cx from 'classnames';
import { FormattedMessage, defineMessages, useIntl } from 'react-intl';

import Cell from '@plone/volto/components/manage/Blocks/Table/Cell';
import { Field, Icon } from '@plone/volto/components';
import { injectLazyLibs } from '@plone/volto/helpers/Loadable/Loadable';
import { useClient } from '@plone/volto/hooks';

import rowBeforeSVG from '@plone/volto/icons/row-before.svg';
import rowAfterSVG from '@plone/volto/icons/row-after.svg';
import colBeforeSVG from '@plone/volto/icons/column-before.svg';
import colAfterSVG from '@plone/volto/icons/column-after.svg';
import rowDeleteSVG from '@plone/volto/icons/row-delete.svg';
import colDeleteSVG from '@plone/volto/icons/column-delete.svg';

const getId = () => Math.floor(Math.random() * Math.pow(2, 24)).toString(32);

const valueToDraft = (value) => ({
  blocks: [
    {
      data: {},
      depth: 0,
      entityRanges: [],
      inlineStyleRanges: [],
      key: getId(),
      text: value,
      type: 'unstyled',
    },
  ],
  entityMap: {},
});

const emptyCell = (type) => ({
  key: getId(),
  type: type || 'data',
  value: valueToDraft(''),
});

const emptyRow = (cells) => ({
  key: getId(),
  cells: map(cells, () => emptyCell()),
});

const initialTable = () => ({
  fixed: true,
  compact: false,
  basic: false,
  celled: true,
  inverted: false,
  striped: false,
  rows: [
    {
      key: getId(),
      cells: [
        {
          key: getId(),
          type: 'header',
          value: valueToDraft(''),
        },
        {
          key: getId(),
          type: 'header',
          value: valueToDraft(''),
        },
      ],
    },
    {
      key: getId(),
      cells: [
        {
          key: getId(),
          type: 'data',
          value: valueToDraft(''),
        },
        {
          key: getId(),
          type: 'data',
          value: valueToDraft(''),
        },
      ],
    },
  ],
});

const messages = defineMessages({
  insertRowBefore: {
    id: 'Insert row before',
    defaultMessage: 'Insert row before',
  },
  insertRowAfter: {
    id: 'Insert row after',
    defaultMessage: 'Insert row after',
  },
  deleteRow: {
    id: 'Delete row',
    defaultMessage: 'Delete row',
  },
  insertColBefore: {
    id: 'Insert col before',
    defaultMessage: 'Insert col before',
  },
  insertColAfter: {
    id: 'Insert col after',
    defaultMessage: 'Insert col after',
  },
  deleteCol: {
    id: 'Delete col',
    defaultMessage: 'Delete col',
  },
  fixed: {
    id: 'Fixed width table cells',
    defaultMessage: 'Fixed width columns',
  },
  compact: {
    id: 'Make the table compact',
    defaultMessage: 'Reduce cell padding',
  },
  basic: {
    id: 'Reduce complexity',
    defaultMessage: 'Minimalistic table design',
  },
  celled: {
    id: 'Divide each row into separate cells',
    defaultMessage: 'Add border to inner columns',
  },
  striped: {
    id: 'Stripe alternate rows with color',
    defaultMessage: 'Alternate row background color',
  },
  headerCell: {
    id: 'Header cell',
    defaultMessage: 'Header cell',
  },
});

const Edit = (props) => {
  const {
    data,
    block,
    onChangeBlock,
    draftJs,
    onAddBlock,
    onSelectBlock,
    index,
  } = props;
  const intl = useIntl();
  const isClient = useClient();
  const [selected, setSelected] = useState({
    row: 0,
    cell: 0,
  });

  const { convertToRaw } = draftJs;

  useEffect(() => {
    if (!data.table) {
      onChangeBlock(block, {
        ...data,
        table: initialTable(),
      });
    }
  }, [data, block, onChangeBlock]);

  useEffect(() => {
    if (!data.table) {
      onChangeBlock(block, {
        ...data,
        table: initialTable(),
      });
    }
  }, [data, block, onChangeBlock]);

  const onSelectCell = (row, cell) => {
    setSelected({
      row: row,
      cell: cell,
    });
  };

  const onChangeCell = (row, cell, editorState) => {
    const table = { ...data.table };
    table.rows[row].cells[cell].value = convertToRaw(
      editorState.getCurrentContent(),
    );
    onChangeBlock(block, {
      ...data,
      table,
    });
  };

  const toggleCellType = () => {
    const table = { ...data.table };
    let type = table.rows[selected.row].cells[selected.cell].type;
    table.rows[selected.row].cells[selected.cell].type =
      type === 'header' ? 'data' : 'header';
    onChangeBlock(block, {
      ...data,
      table,
    });
  };

  const onInsertRowBefore = (e) => {
    e.preventDefault();
    e.stopPropagation();
    const table = data.table;
    onChangeBlock(block, {
      ...data,
      table: {
        ...table,
        rows: [
          ...table.rows.slice(0, selected.row),
          emptyRow(table.rows[0].cells),
          ...table.rows.slice(selected.row),
        ],
      },
    });
    setSelected({
      row: selected.row + 1,
      cell: selected.cell,
    });
  };

  const onInsertRowAfter = (e) => {
    e.preventDefault();
    e.stopPropagation();
    const table = data.table;
    onChangeBlock(block, {
      ...data,
      table: {
        ...table,
        rows: [
          ...table.rows.slice(0, selected.row + 1),
          emptyRow(table.rows[0].cells),
          ...table.rows.slice(selected.row + 1),
        ],
      },
    });
  };

  const onInsertColBefore = (e) => {
    e.preventDefault();
    e.stopPropagation();
    const table = data.table;
    onChangeBlock(block, {
      ...data,
      table: {
        ...table,
        rows: map(table.rows, (row, index) => ({
          ...row,
          cells: [
            ...row.cells.slice(0, selected.cell),
            emptyCell(table.rows[index].cells[selected.cell].type),
            ...row.cells.slice(selected.cell),
          ],
        })),
      },
    });
    setSelected({
      row: selected.row,
      cell: selected.cell + 1,
    });
  };

  const onInsertColAfter = (e) => {
    e.preventDefault();
    e.stopPropagation();
    const table = data.table;
    onChangeBlock(block, {
      ...data,
      table: {
        ...table,
        rows: map(table.rows, (row, index) => ({
          ...row,
          cells: [
            ...row.cells.slice(0, selected.cell + 1),
            emptyCell(table.rows[index].cells[selected.cell].type),
            ...row.cells.slice(selected.cell + 1),
          ],
        })),
      },
    });
  };

  const onDeleteCol = (e) => {
    e.preventDefault();
    e.stopPropagation();
    const table = data.table;

    if (selected.cell === table.rows[0].cells.length - 1) {
      setSelected({
        row: selected.row,
        cell: selected.cell - 1,
      });
    }

    onChangeBlock(block, {
      ...data,
      table: {
        ...table,
        rows: map(table.rows, (row) => ({
          ...row,
          cells: remove(row.cells, (cell, index) => index !== selected.cell),
        })),
      },
    });
  };

  const onDeleteRow = (e) => {
    e.preventDefault();
    e.stopPropagation();
    const table = data.table;

    if (selected.row === table.rows.length - 1) {
      setSelected({
        row: selected.row - 1,
        cell: selected.cell,
      });
    }

    onChangeBlock(block, {
      ...data,
      table: {
        ...table,
        rows: remove(table.rows, (row, index) => index !== selected.row),
      },
    });
  };

  const toggleBool = (value) => {
    const table = data.table;
    onChangeBlock(block, {
      ...data,
      table: {
        ...table,
        [value]: !table[value],
      },
    });
  };

  const toggleFixed = () => {
    toggleBool('fixed');
  };

  const toggleCompact = () => {
    toggleBool('compact');
  };

  const toggleBasic = () => {
    toggleBool('basic');
  };

  const toggleCelled = () => {
    toggleBool('celled');
  };

  const toggleStriped = () => {
    toggleBool('striped');
  };

  if (!isClient) {
    return <div />;
  }

  return (
    <div className={cx('block table', { selected: props.selected })}>
      {props.selected && (
        <div className="toolbar">
          <Button.Group>
            <Button
              icon
              basic
              onClick={onInsertRowBefore}
              title={intl.formatMessage(messages.insertRowBefore)}
              aria-label={intl.formatMessage(messages.insertRowBefore)}
            >
              <Icon name={rowBeforeSVG} size="24px" />
            </Button>
          </Button.Group>
          <Button.Group>
            <Button
              icon
              basic
              onClick={onInsertRowAfter}
              title={intl.formatMessage(messages.insertRowAfter)}
              aria-label={intl.formatMessage(messages.insertRowAfter)}
            >
              <Icon name={rowAfterSVG} size="24px" />
            </Button>
          </Button.Group>
          <Button.Group>
            <Button
              icon
              basic
              onClick={onDeleteRow}
              disabled={data.table && data.table.rows.length === 1}
              title={intl.formatMessage(messages.deleteRow)}
              aria-label={intl.formatMessage(messages.deleteRow)}
            >
              <Icon name={rowDeleteSVG} size="24px" />
            </Button>
          </Button.Group>
          <Button.Group>
            <Button
              icon
              basic
              onClick={onInsertColBefore}
              title={intl.formatMessage(messages.insertColBefore)}
              aria-label={intl.formatMessage(messages.insertColBefore)}
            >
              <Icon name={colBeforeSVG} size="24px" />
            </Button>
          </Button.Group>
          <Button.Group>
            <Button
              icon
              basic
              onClick={onInsertColAfter}
              title={intl.formatMessage(messages.insertColAfter)}
              aria-label={intl.formatMessage(messages.insertColAfter)}
            >
              <Icon name={colAfterSVG} size="24px" />
            </Button>
          </Button.Group>
          <Button.Group>
            <Button
              icon
              basic
              onClick={onDeleteCol}
              disabled={data.table && data.table.rows[0].cells.length === 1}
              title={intl.formatMessage(messages.deleteCol)}
              aria-label={intl.formatMessage(messages.deleteCol)}
            >
              <Icon name={colDeleteSVG} size="24px" />
            </Button>
          </Button.Group>
        </div>
      )}
      {data.table && (
        <Table
          fixed={data.table.fixed}
          compact={data.table.compact}
          basic={data.table.basic ? 'very' : false}
          celled={data.table.celled}
          inverted={data.table.inverted}
          striped={data.table.striped}
        >
          <Table.Body>
            {map(data.table.rows, (row, rowIndex) => (
              <Table.Row key={row.key}>
                {map(row.cells, (cell, cellIndex) => (
                  <Table.Cell
                    key={cell.key}
                    as={cell.type === 'header' ? 'th' : 'td'}
                    className={
                      rowIndex === selected.row &&
                      cellIndex === selected.cell &&
                      props.selected
                        ? 'selected'
                        : ''
                    }
                  >
                    <Cell
                      editable={props.editable}
                      value={cell.value}
                      row={rowIndex}
                      cell={cellIndex}
                      onSelectCell={onSelectCell}
                      selected={
                        rowIndex === selected.row && cellIndex === selected.cell
                      }
                      isTableBlockSelected={props.selected}
                      onAddBlock={onAddBlock}
                      onSelectBlock={onSelectBlock}
                      onChange={onChangeCell}
                      index={index}
                      disableNewBlocks={data?.disableNewBlocks}
                    />
                  </Table.Cell>
                ))}
              </Table.Row>
            ))}
          </Table.Body>
        </Table>
      )}
      {props.selected && isClient && (
        <Portal node={document.getElementById('sidebar-properties')}>
          <Form
            method="post"
            onSubmit={(event) => {
              event.preventDefault();
              event.stopPropagation();
            }}
          >
            <Segment secondary attached>
              <FormattedMessage id="Table" defaultMessage="Table" />
            </Segment>
            <Segment attached>
              <Field
                id="fixed"
                title={intl.formatMessage(messages.fixed)}
                type="boolean"
                value={data.table && data.table.fixed}
                onChange={() => toggleFixed()}
              />
              <Field
                id="celled"
                title={intl.formatMessage(messages.celled)}
                type="boolean"
                value={data.table && data.table.celled}
                onChange={toggleCelled}
              />
              <Field
                id="striped"
                title={intl.formatMessage(messages.striped)}
                type="boolean"
                value={data.table && data.table.striped}
                onChange={toggleStriped}
              />
              <Field
                id="compact"
                title={intl.formatMessage(messages.compact)}
                type="boolean"
                value={data.table && data.table.compact}
                onChange={() => toggleCompact()}
              />
              <Field
                id="basic"
                title={intl.formatMessage(messages.basic)}
                type="boolean"
                value={data.table && data.table.basic}
                onChange={toggleBasic}
              />
            </Segment>
            <Segment secondary attached>
              <FormattedMessage id="Cell" defaultMessage="Cell" />
            </Segment>
            <Segment attached>
              <Field
                id="celltype"
                title={intl.formatMessage(messages.headerCell)}
                type="boolean"
                value={
                  data.table &&
                  data.table.rows[selected.row].cells[selected.cell].type ===
                    'header'
                }
                onChange={toggleCellType}
              />
            </Segment>
          </Form>
        </Portal>
      )}
    </div>
  );
};

Edit.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
  detached: PropTypes.bool,
  index: PropTypes.number.isRequired,
  selected: PropTypes.bool.isRequired,
  block: PropTypes.string.isRequired,
  onAddBlock: PropTypes.func.isRequired,
  onInsertBlock: PropTypes.func.isRequired,
  onChangeBlock: PropTypes.func.isRequired,
  onDeleteBlock: PropTypes.func.isRequired,
  onMutateBlock: PropTypes.func.isRequired,
  onFocusPreviousBlock: PropTypes.func.isRequired,
  onFocusNextBlock: PropTypes.func.isRequired,
  onSelectBlock: PropTypes.func.isRequired,
};
Edit.defaultProps = {
  detached: false,
};
export default compose(injectLazyLibs(['draftJs']))(Edit);
