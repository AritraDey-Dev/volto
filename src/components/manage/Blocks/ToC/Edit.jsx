import React from 'react';

import { SidebarPortal } from '@plone/volto/components';
import BlockDataForm from '@plone/volto/components/manage/Form/BlockDataForm';

import TableOfContentsSchema from './Schema';
import View from './View';

const Edit = (props) => {
  const { selected, data, block, onChangeBlock } = props;
  const schema = TableOfContentsSchema(props);

  return (
    <>
      <View {...props} mode="edit" />

      <SidebarPortal selected={selected}>
        <BlockDataForm
          schema={schema}
          title={schema.title}
          onChangeField={(id, value) => {
            onChangeBlock(block, {
              ...data,
              [id]: value,
            });
          }}
          onChangeBlock={onChangeBlock}
          formData={data}
          block={block}
        />
      </SidebarPortal>
    </>
  );
};

export default Edit;
