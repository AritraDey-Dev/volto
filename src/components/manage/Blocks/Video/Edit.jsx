/**
 * Edit video block.
 * @module components/manage/Blocks/Title/Edit
 */

import React from 'react';
import PropTypes from 'prop-types';
import { defineMessages } from 'react-intl';
import cx from 'classnames';

import { SidebarPortal, VideoSidebar } from '@plone/volto/components';
import Body from '@plone/volto/components/manage/Blocks/Video/Body';
import { withBlockExtensions } from '@plone/volto/helpers';
import MediaWidget from '@plone/volto/components/manage/Widgets/MediaSelectWidget';

const messages = defineMessages({
  VideoFormDescription: {
    id: 'Specify a youtube video or playlist url',
    defaultMessage: 'Specify a youtube video or playlist url',
  },
  VideoBlockInputPlaceholder: {
    id: 'Type a Video (YouTube, Vimeo or mp4) URL',
    defaultMessage: 'Type a Video (YouTube, Vimeo or mp4) URL',
  },
});

/**
 * Edit video block function.
 * @function Edit
 */
const Edit = (props) => {
  const { data, editable, block, onChangeBlock, selected } = props;
  return (
    <div
      className={cx(
        'block video align',
        {
          selected: selected,
          center: !Boolean(data.align),
        },
        data.align,
      )}
    >
      {data.url ? (
        <Body data={data} isEditMode={true} />
      ) : (
        <div>
          {editable && (
            <MediaWidget
              inline
              mode="video"
              id="url"
              title="Source"
              block={block}
              onChange={(id, value) => {
                onChangeBlock(block, {
                  ...data,
                  [id]: value,
                });
              }}
              handlesErrors={false}
            />
          )}
        </div>
      )}
      <SidebarPortal selected={selected}>
        <VideoSidebar {...props} />
      </SidebarPortal>
    </div>
  );
};

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
Edit.propTypes = {
  selected: PropTypes.bool.isRequired,
  block: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
  data: PropTypes.objectOf(PropTypes.any).isRequired,
  onChangeBlock: PropTypes.func.isRequired,
  onSelectBlock: PropTypes.func.isRequired,
  onDeleteBlock: PropTypes.func.isRequired,
  onFocusPreviousBlock: PropTypes.func.isRequired,
  onFocusNextBlock: PropTypes.func.isRequired,
  handleKeyDown: PropTypes.func.isRequired,
};

export default withBlockExtensions(Edit);
