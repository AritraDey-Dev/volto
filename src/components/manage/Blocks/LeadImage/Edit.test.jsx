import React from 'react';
import { Provider } from 'react-intl-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import Edit from './Edit';

const mockStore = configureStore();

test('renders an edit Lead Image block component', () => {
  const store = mockStore({
    intl: {
      locale: 'en',
      messages: {},
    },
  });
  const component = renderer.create(
    <Provider store={store}>
      <Edit
        data={{}}
        properties={{
          image: {
            download: 'http://127.0.0.1:3000/image.png',
            width: 400,
            height: 400,
            scales: {
              preview: {
                download:
                  'http://127.0.0.1:3000/image.png/@@images/image/image-400.png',
                width: 400,
                height: 400,
              },
            },
          },
        }}
        selected={false}
        block="1234"
        content={{}}
        request={{
          loading: false,
          loaded: false,
        }}
        pathname="/news"
        onChangeBlock={() => {}}
        onChangeField={() => {}}
        index={1}
        openObjectBrowser={() => {}}
      />
    </Provider>,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
