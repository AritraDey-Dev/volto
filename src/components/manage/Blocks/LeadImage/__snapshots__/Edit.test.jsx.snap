// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an edit Lead Image block component 1`] = `
<div
  className="block image align center"
>
  <img
    alt=""
    className="responsive"
    imageField="image"
    item={
      Object {
        "image": Object {
          "download": "http://127.0.0.1:3000/image.png",
          "height": 400,
          "scales": Object {
            "preview": Object {
              "download": "http://127.0.0.1:3000/image.png/@@images/image/image-400.png",
              "height": 400,
              "width": 400,
            },
          },
          "width": 400,
        },
      }
    }
  />
</div>
`;
